src/common/meta/src/wal/kafka/topic_selector.rs

```diff
@@ -29,23 +31,56 @@ pub enum SelectorType {
 /// Controls topic selection.
 pub(super) trait TopicSelector: Send + Sync {
     /// Selects a topic from the topic pool.
-    fn select(&self, topic_pool: &[Topic]) -> &Topic;
+    fn select<'a>(&'a self, topic_pool: &'a [Topic]) -> &Topic;
 }
 
+/// Arc wrapper of TopicSelector.
 pub(super) type TopicSelectorRef = Arc<dyn TopicSelector>;
 
 /// A topic selector with the round-robin strategy, i.e. selects topics in a round-robin manner.
-pub(super) struct RoundRobinTopicSelector;
+#[derive(Default)]
+pub(super) struct RoundRobinTopicSelector {
+    cursor: AtomicUsize,
+}
 
 impl RoundRobinTopicSelector {
-    /// Creates a new round-robin topic selector.
-    pub(super) fn new() -> Self {
-        todo!()
+    // The cursor in the round-robin selector is not persisted which may break the round-robin strategy cross crashes.
+    // Introducing a shuffling strategy may help mitigate this issue.
+    pub fn with_shuffle() -> Self {
+        let mut this = Self::default();
+        let offset = rand::thread_rng().gen::<usize>() % usize::MAX;
```

Comment:
I think we can generate the offset from a small range of random numbers.

Reply:
The offset is now generated from range 0..64.